packages/tooltip/src/use-tooltip.ts

```diff
@@ -246,3 +257,13 @@ export function useTooltip(props: UseTooltipProps = {}) {
 }
 
 export type UseTooltipReturn = ReturnType<typeof useTooltip>
+
+const closeEventName = "chakra-ui:close-tooltip"
+
+function useCloseEvent(close: () => void) {
+  useEffect(() => {
+    document.addEventListener(closeEventName, close)
+    return () => document.removeEventListener(closeEventName, close)
+  }, [close])
+  return () => document.dispatchEvent(new CustomEvent(closeEventName))
+}
```

Comment:
Great work! Can we ensure we use the correct reference to `window` and `document` for the tooltip?

doc => `ref.current.ownerDocument` 
win => `doc.defaultView`

Reply:
Fixed 19ba261b1d3ed1d50ed794583d90f7dd4b7d20bd

Reply:
I don't think that's the correct fix. Here's what you can consider

1. Allow useCloseEvent to accept a ref (same ref that the tooltip creates)
2. Within the useEffect of the useCloseEvent and useTooltip, read the `doc` and `win` from there

This way, we can avoid the useForceUpdate completely

Reply:
Thanks for your review! I fixed not to use `useForceUpdate`. af89344346a1318480710c6df1ab5a6d9eedda61